ingestion-beam/src/main/java/com/mozilla/telemetry/schemas/SchemaStore.java

```diff
@@ -0,0 +1,145 @@
+    ensureSchemasLoaded();
+    if (attributes == null) {
+      throw new SchemaNotFoundException("No schema for message with null attributeMap");
+    }
+    // This is the path provided by mozilla-pipeline-schemas
+    final String path = StringSubstitutor.replace("${document_namespace}/${document_type}/"
+        + "${document_type}.${document_version}.schema.json", attributes);
+    return getSchema(path);
+  }
+
+  /** Returns true if we found at least one schema with the given doctype and namespace. */
+  public boolean docTypeExists(String namespace, String docType) {
+    ensureSchemasLoaded();
+    if (namespace == null || docType == null) {
+      return false;
+    } else {
+      return dirs.contains(namespace + "/" + docType);
+    }
+  }
+
+  /** Returns true if we found at least one schema with matching the given attributes. */
+  public boolean docTypeExists(Map<String, String> attributes) {
+    String namespace = attributes.get("document_namespace");
+    String docType = attributes.get("document_type");
+    return docTypeExists(namespace, docType);
+  }
+
+  ////////
+
+  private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(SchemaStore.class);
+
+  private final ValueProvider<String> schemasLocation;
+  private transient Map<String, Schema> schemas;
+  private transient Set<String> dirs;
+
+  private SchemaStore(ValueProvider<String> schemasLocation) {
+    this.schemasLocation = schemasLocation;
+  }
+
+  @VisibleForTesting
+  public int numLoadedSchemas() {
+    ensureSchemasLoaded();
+    return schemas.size();
+  }
+
+  private void loadAllSchemas() throws SchemaNotFoundException, IOException {
+    final Map<String, Schema> tempSchemas = new HashMap<>();
+    final Set<String> tempDirs = new HashSet<>();
+    InputStream inputStream;
+    try {
+      Metadata metadata = FileSystems.matchSingleFileSpec(schemasLocation.get());
+      ReadableByteChannel channel = FileSystems.open(metadata.resourceId());
+      inputStream = Channels.newInputStream(channel);
+    } catch (IOException e) {
+      throw new IOException("Exception thrown while fetching from configured schemasLocation", e);
+    }
+    try (InputStream bi = new BufferedInputStream(inputStream);
+        InputStream gzi = new GzipCompressorInputStream(bi);
+        ArchiveInputStream i = new TarArchiveInputStream(gzi);) {
+      ArchiveEntry entry = null;
+      while ((entry = i.getNextEntry()) != null) {
+        if (!i.canReadEntryData(entry)) {
+          LOG.warn("Unable to read tar file entry: " + entry.getName());
+          continue;
+        }
+        String[] components = entry.getName().split("/");
+        if (components.length > 2 && "schemas".equals(components[1])) {
+          String name = String.join("/", Arrays.copyOfRange(components, 2, components.length));
+          if (entry.isDirectory()) {
+            tempDirs.add(name);
+          } else if (name.endsWith(".json")) {
```

Comment:
I think that @acmiyaguchi is going to be outputting `bigquery` schemas as `.bigquery.json` files, so this may cause issues when that is merged.